### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by zschira**
This pull request adds the ability to parse XBRL taxonomies from a zipfile. Arelle already has this capability, so it's a fairly small change to tell Arelle the location of the zipfile, and the path to the taxonomy entry point within it.

### CODE REVIEW

File: src/ferc_xbrl_extractor/arelle_interface.py [hunk 1]
```diff
@@ -28,3 +28,17 @@ def load_taxonomy(path: str):
...
+
+def load_taxonomy_from_archive(filepath: str, archive_path: str):
+    """Load an XBRL taxonomy from a zipfile archive.
+
+    Args:
+        filepath: Path to zipfile on disc.
+        archive_path: Relative path to taxonomy entry point within archive.
+    """
+    # Create arelle FileSource object
+    print(f"archive_path={archive_path}")
```
Comment (bendnorman) [inline src/ferc_xbrl_extractor/arelle_interface.py:41]: Can this be replaced with a logger method?

**Review by bendnorman — CHANGES_REQUESTED** [review]
Looks good! So before these changes, the `xbrl_extract ` command extracted XBRL filing directly from the FERC API. Now, you can point `--archive-path` to the archived XBRL filings in zipfile produced by your scrapers?

File: src/ferc_xbrl_extractor/arelle_interface.py (hunk 1 above)
Comment (zschira) [reply src/ferc_xbrl_extractor/arelle_interface.py:41]: Oh yeah I'd just put this in for debugging, so I just removed it

**Comment by zschira** [comment]
Previously it was getting the taxonomies from FERC directly, but it was parsing taxonomies from disk. This option allows it to parse taxonomies from a local zipfile.

**Review by bendnorman — APPROVED** [review]

**Pull request merged**